bool det_flag
---
bool success
string message